bloom-housing/bloom
# Prisma Translation service

## Issue Overview

This PR addresses #3496 


## Description

Add the translation service to the new Prisma backend. This does not require a controller so it is just the service that translates listings and gets the translations from the DB

## How Can This Be Tested/Reviewed?

This can only be partially tested as the one service endpoint is currently not being used.

Testing the `translateListing` endpoint:
- When running the backend locally, you can hit the get listing endpoint from swagger: http://localhost:3101/api#/listings/retrieve.
- Get a listing id from the db and hit the above endpoint with that listing id and a non-english language (e.g `es`)
- All of the translatable fields on the listing should be translated to the selected language in the output.
- Also these new translations should show up in the `generated_listing_translations` table. If you hit the endpoint again a call to google should not happen (would require console logging added to verify).


## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
